frontend/viewer/src/lib/components/reorderer/reorderer-item-list.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts" module>
+  export type ReordererItemListProps = Omit<HTMLAttributes<HTMLElement>, 'onchange'> & {
+    id?: string;
+  };
+</script>
+
+<script lang="ts" generics="T">
+  import {watch} from 'runed';
+  import {Icon} from '$lib/components/ui/icon';
+  import * as DropdownMenu from '$lib/components/ui/dropdown-menu';
+  import type {HTMLAttributes} from 'svelte/elements';
+  import {mergeProps} from 'bits-ui';
+  import {cn} from '$lib/utils';
+  import {useReordererItemList} from './reorderer.svelte';
+
+  let {
+    ...rest
+  } : ReordererItemListProps = $props();
+
+  const itemListState = useReordererItemList<T>();
+  const {item, items, getDisplayName, onchange} = $derived(itemListState);
```

Comment:
I'm not totally sure what you're going for here, but this `$dferived` doesn't do anything since `itemListState` isn't a state object. It might be deeply reactive but then deconstructing it breaks that also.

Reply:
Actually, it seems the problem was that changes weren't even making it into the `itemListState` object.
I was just passing in a snapshot of the initial values and never updating it.